From: Romain Perier <romain.perier@gmail.com>
Date: Wed, 16 Oct 2019 19:03:00 +0200
Subject: ARM: dts: bcm283x: Correct vchiq compatible string

This allows VCHIQ to determine the correct cache line size, use the new
"brcm,bcm2836-vchiq" compatible string on BCM2836. It is based on commit
499770 ("ARM: dts: bcm283x: Correct vchiq compatible string")

Signed-off-by: Romain Perier <romain.perier@gmail.com>
---
 arch/arm/boot/dts/bcm2835-rpi.dtsi | 2 +-
 arch/arm/boot/dts/bcm2836-rpi.dtsi | 6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/bcm2836-rpi.dtsi

--- a/arch/arm/boot/dts/bcm2835-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2835-rpi.dtsi
@@ -30,7 +30,7 @@
 			#power-domain-cells = <1>;
 		};
 
-		mailbox@7e00b840 {
+		vchiq: mmailbox@7e00b840 {
 			compatible = "brcm,bcm2835-vchiq";
 			reg = <0x7e00b840 0x3c>;
 			interrupts = <0 2>;
--- /dev/null
+++ b/arch/arm/boot/dts/bcm2836-rpi.dtsi
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "bcm2835-rpi.dtsi"
+
+&vchiq {
+	compatible = "brcm,bcm2836-vchiq", "brcm,bcm2835-vchiq";
+};
